### DISCUSSION TITLE
Outreachy Discussions - 2022 Round 1

### DISCUSSION THREAD

**Discussion by MridulS** [networkx/networkx#5396] [General]
Hello all!

Please use this discussion page to discuss everything for Outreachy.
If you have any questions or doubts about the processes please put in a reply to this discussion post.

**Comment by AliciaElizabeta** [comment]
Hi, I'm Alicia. I hope we can work well together.

**Reply by HASSAN1A** [reply]
Hi,
 I am Hassan Juma from from Nairobi Kenya happy to be here and really want to work on the project

**Comment by anna4j** [comment]
Hi, I'm Anushka. Here to learn and contribute!

**Comment by cherotich** [comment]
Hi, My name is Caroline. I am super happy to collaborate with you guys. Cheers.

**Comment by NikitaSharma1** [comment]
Hi, I am Nikita. I am so excited to learn and work with you. Looking forward to contribute!

**Comment by singhmansi25** [comment]
Hi !!
I'm Mansi. I'm here to explore, learn and experience. Looking forward to a great journey of contribution.

**Comment by 0ddoes** [comment]
Hey all! My name is Izza. I am from India. Really looking forward to this contribution period. Can you please guide me on how I can start? Shall I look at the Good first issues initially? Do the issues have to be assigned to us to start working or can we open a PR when we are ready?

**Comment by aliveevie** [comment]
Hello, My Name is Ibrahim Abdulkarim, from Nigeria. Nice to meet you here

**Comment by Astroakanksha24** [comment]
Hello, this is Akanksha from India. Looking forward to contributing!!

**Comment by parikshit14** [comment]
Hi Team,

I am Parikshit Singh Rathore, I am a senior year undergraduate studying computer science and engineering. I have been contributing to open-source for a year and a half and would love to contribute & be part of this team for this summer through outreachy.

Would be delighted to discuss project topics further and start working on them.
Thanks

**Comment by Teresayyyy** [comment]
Hi, I'm Teresa,  nice to meet you all and looking forward to contributing and working with you!

**Comment by Lukong123** [comment]
Hello everyone, I am Lukong Anne and it's a privilege to be here. Looking forward to contributing to the community!!

**Comment by martwebber** [comment]
Hello, I am Martin. I'm excited to be part of this community.

**Comment by MridulS** [comment]
Hello all! I'm super excited to see so many folks interested in contributing to NetworkX. 

As a first step: applicants who haven't used networkx before should first [install networkx](https://networkx.org/documentation/latest/install.html) on their machines and follow our [introductory tutorial](https://networkx.org/documentation/latest/tutorial.html). This will give applicants a better overview of our codebase and API, which will help the applicant create a strong application.

Once applicants are comfortable with using networkx, they should go through our [Contributor Guide](https://networkx.org/documentation/latest/developer/contribute.html) to setup our local development environment for networkx, if the applicant is interested in working on specific issues they can comment on them ([Good First issues](https://github.com/networkx/networkx/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+First+Issue%22)). Applicants can also pick *any* other [open issue](https://github.com/networkx/networkx/issues) if they like or even open a new issue/discussion if there is something in our documentation or codebase that doesn't make sense :)

The applicant should also go through the [New Contributor FAQs](https://networkx.org/documentation/latest/developer/new_contributor_faq.html) as it may answer some of their questions.

As the project will deal with creating new notebooks for our NX-guides please do go through current notebooks at  https://networkx.org/nx-guides/.

Feel free to post here if you have any other questions or issues!

Excited for the summer! :)

**Reply by deepthi1107** [reply]
Thanks, MridulS for the information.

**Reply by esticojerry** [reply]
Thanks for the information

**Comment by MridulS** [comment]
Hello all!

Just wanted to post another update. We have created another repository https://github.com/networkx/outreachy which will have explicit tasks for Outreachy applicants during the contribution period.

To participate with NetworkX as an Outreachy intern applicants are expected to go through multiple tasks in that repository. All tasks are presented as [issues](https://github.com/networkx/outreachy/issues). We will add a couple of more tasks to this repository.

Everyone is expected to start with https://github.com/networkx/outreachy/issues/2.

Feel free to post here if you have any other questions!

**Reply by aliveevie** [reply]
@MridulS I have done all the task on that repo, please I'm expecting feedback on my tasks.

**Comment by Inny05** [comment]
Hello everyone,
My name is Inny, and I am super excited to be here. I look forward to learning and improving on my skills.

**Comment by angelnagaba** [comment]
Hello Everyone my name is Angel. super excited to contribute to this project

**Comment by krunal-gamit** [comment]
Hello everyone! 
I am Krunal Gamit, Outreachy applicant. An Open-Source enthusiast & a junior pursuing my bachelor's in Computer Science & Engineering. I am passionate about Web Dev, Machine Learning, DevOps & I enjoy learning new things. 
Looking forward to contributing to this project.

**Comment by soumya20134** [comment]
Hello All, I am Soumya. Super pumped to be part of this community and contribute!

**Comment by deepthi1107** [comment]
Hello all, I am Deepthi, excited to be a part of great summer contributions. Looking forward to learning, exploring, and expanding my skills.
I would like to contribute, please do guide me.

**Comment by Adebesindedayo** [comment]
I am Adedayo,hoping to make substantial contribution

**Comment by shivani6320** [comment]
Hello Everyone!! I'm Shivani - an outreachy applicant🤓. I am super excited to be part of this project and learn from the mentors and fellow community members. Looking forward to making meaningful contributions.✨

**Comment by olawale-ahmed** [comment]
Hello, everyone. I am Olawale Ahmed, Outreachy applicant. I am ready to contribute, learn and collaborate on Networkx library. I have some experience with Python and Jupyter notebooks.

**Comment by NishithaThimmappa** [comment]
Hi, everyone I am Nishitha from Indian Institute of Technology, Outreachy applicant. Looking forward to making meaningful contributions.

**Comment by Beatrice1d** [comment]
Hello Everyone, 
I am Beatrice, an Outreachy applicant from Nigeria. Looking forward to awesome contributions and interactions

**Comment by FAIZ113** [comment]
Hello everyone,
My name is Amina Faiz- an Outreachy applicant from India. I am a first-year graduate student studying computer science. I am super excited to contribute to this community. I look forward to learning and improving my skills.

**Reply by deepthi1107** [reply]
Welcome Faiz

**Comment by ManasviGoyal** [comment]
Hi Everyone,

I am Manasvi Goyal, a prefinal year student of B.Tech. I started my open-source journey last year and have contributed to various communities. I am hoping to get an opportunity to learn more about open source and this project during this contribution period and would like to continue my contribution as a part of this team even after it to make meaningful contributions to this project.

Looking forward to learning many things.

Thank you

**Comment by nandini2511** [comment]
Hello everyone! 
I'm Nandini from India. I'm a sophomore pursuing a B.Tech in Information Technology. I'm excited to learn and contribute to this project!

**Comment by Beatrice1d** [comment]
@MridulS 

Please I have a challenge with setting up the environment and would appreciate a pointer on what to do differently. 

Following the steps outlined in the contributory guidelines have forked, cloned the repo, and have added the upstream repo.
However, I am stuck trying to set up my build environment. 

With the command "python -m venv networkx-dev",  I created an empty directory inside my Netoworkx directory 
However, when I tried "source networkx-dev/bin/activate", I got an error that say "bash: networkx-dev/bin/activate: No such file or directory"

Note -1: I tried creating the networkx-dev folder outside the networkx folder but got the same error above.
Note -2: I am using a Windows PC for my tasks.

**Reply by shivani6320** [reply]
Hey @Beatrice1d , have you tried using  "networkx-dev/scripts/activate" ?
because I was having the same error & above mentioned code worked for me.(I'm also using Windows PC)

**Comment by yashre-bh** [comment]
Hello!
Shreya Bhagat here, a Sophomore at IIT Roorkee, India. super pumped to learn and contribute. can someone guide me on how to get started?

**Reply by MridulS** [reply]
Thanks for your interest in contributing to NetworkX! Please through https://github.com/networkx/networkx/discussions/5396#discussioncomment-2447770 and https://github.com/networkx/networkx/discussions/5396#discussioncomment-2449797, if you have any other questions feel free to post them here.

**Comment by Saumay1** [comment]
Hello everyone!! This is Saumya, looking forward to learning and contributing.

**Comment by JyotiKM29** [comment]
Hello everyone ! I 'm Jyoti . Looking forward learn and Contribute .

**Comment by deepthi1107** [comment]
Hello @MridulS, while plotting the degree centrality of random erdos-renyi graph and barabasi_albert_graph, the distribution of hist is uniform, and even though I have changed the number of bins number the result is the same... Is I am going on the right path of visualizing the hist plot (distribution).

**Reply by dschult** [reply]
Hi @deepthi1107 
You should probably create the PR using what you have got and then ask this question inside that PR.  That way we can see what you are doing and try to give pointers. Without that extra information it is hard to know what is going wrong.  This is the way we look at each other's work with Github.  Create a PR with what we've done so far, and then leave comments/questions in the conversation about that PR.   (probably also good to mention in the PR that this addresses issue #8)

**Reply by deepthi1107** [reply]
sure, thanks i definitely follow the steps and ask my quries through that.

**Reply by deepthi1107** [reply]
Actually, I was plotting the hist plot in the wrong way, and figured out how to do it.

**Comment by deepthi1107** [comment]
@MridulS, I have made changes in my PR, guide me if I am wrong.

**Comment by deepthi1107** [comment]
@MridulS, I have made PR and provide me feedback

**Reply by deepthi1107** [reply]
Anyone who know FOSS community channel link, share with me..
Thank you.

**Reply by MridulS** [reply]
Hello @deepthi1107! thanks for the nudge but as you can see from https://github.com/networkx/outreachy/pulls there are lot of open pull requests and reviewing the Jupyter Notebooks take more time than for example task 1. So please give us enough time to go through the submissions.

For the FOSS community channel link are you looking for something specific to NetworkX or something in general related to FOSS?

**Reply by deepthi1107** [reply]
Both of the links @MridulS.

**Reply by MridulS** [reply]
We don't really have anything specific for NetworkX. We do have [a mailing list](https://groups.google.com/forum/#!forum/networkx-discuss) for announcements and some discussions. For a FOSS specific one maybe you can also checkout [FOSS United](https://fossunited.org), it's a group based out of India which works on the FOSS ecosystem in India, they have a telegram group too.

**Reply by deepthi1107** [reply]
Thanks @MridulS

**Comment by aliveevie** [comment]
@MridulS, Please what will be our contribution URL for submission to outreachy portal for review

**Reply by MridulS** [reply]
It should be the links to pull requests you make to the networkx/outreachy repo, for example https://github.com/networkx/outreachy/pull/56

**Reply by aliveevie** [reply]
If I submit this link, is it ok: https://github.com/networkx/outreachy/tree/main/2022-round-1/aliveevie

**Reply by MridulS** [reply]
Should be fine.

**Comment by Saumay1** [comment]
Hi, @MridulS!
I am trying to to set up build environment using conda  but am stuck in this 


PYTHONPATH=. pytest networkx

![Screenshot (294)](https://user-images.githubusercontent.com/46322717/161975003-1bb9488f-d776-43b0-ae7f-414ad221d17d.png)

I am using Windows.
Could u please help.

**Reply by singhmansi25** [reply]
@Saumay1 
I faced same issue.
You can simply use **pytest networkx** to test if you are using anaconda prompt.
After this wait for some time, so that it executes successfully.

**Reply by Saumay1** [reply]
Thanks a lot @singhmansi25 :)
It worked, I was stuck in this for so long.

**Reply by singhmansi25** [reply]
@Saumay1 
Happy to help

**Comment by unna97** [comment]
Hey!
I am a little late to introduce myself. I am Unnati. I am from India. Super excited to be here!

**Reply by singhmansi25** [reply]
No problem @unna97 
You can check the issues to start your contribution.
Happy to connect

**Reply by Saumay1** [reply]
Welcome @unna97 :)

**Comment by FAIZ113** [comment]
Hi mentor @MridulS!
I am trying to run the test suite locally from the contribution guide but I am getting Value Errors.
I am really stuck. Please guide me where I went wrong and how to get the tests done properly.

![image](https://user-images.githubusercontent.com/60294451/162094950-c8a094db-4520-4fb0-8bf2-b71d05250494.png)

**Comment by Ivy8127** [comment]
Hello, @MridulS  can I still contribute to this project?

**Reply by MridulS** [reply]
Hello @Ivy8127 Contributions are always welcome :) But for outreachy we already have a lot of applicants.

**Comment by aliveevie** [comment]
@MridulS 
Outreachy internship project timeline:
Please work with your mentor to provide a timeline of the work you plan to accomplish on the project and what tasks you will finish at each step. Make sure take into account any time commitments you have during the Outreachy internship round. If you are still working on your contributions and need more time, you can leave this blank and edit your application later.
Please how do we update this information on the outreachy portal.
Thanks

**Reply by MridulS** [reply]
A rough timeline for the pedagogical notebook should follow something like

Week of May 30: Research about the algorithm, dataset or method for the pedagogical notebook 1
Week of June 6: Implementation and Review
Week of June 13: Review

Week of June 20: Research about the algorithm, dataset or method for the pedagogical notebook 2
Week of June 27: Implementation and Review
Week of July 4: Review

Week of July 11: Research about the algorithm, dataset or method for the pedagogical notebook 3
Week of July 18: Implementation and Review
Week of July 25: Review

Week of Aug 1: Research about the algorithm, dataset or method for the pedagogical notebook 4
Week of Aug 8: Implementation and Review
Week of Aug 15: Review
Week of Aug 22: Wrap up of the internship

Feel free to add more or less details to the timeline.

**Comment by Lukong123** [comment]
Hello everyone,
Please sirs I am not able to fetch upstream from my forked repository on Github. I was asked to create a pull request but I don't know how to handle the situation the methods I tried out did not remedy the situation. Sir, I will love to get your advice on this.
Here's a link to my forked repo 
https://github.com/Lukong123/networkx
Please I will appreciate your help in solving this problem.
![pb](https://user-images.githubusercontent.com/70434467/166661146-8ce43b82-6631-43eb-a25f-e657dac276b1.png)

**Comment by dschult** [comment]
I would say that the best approach is to fix it on a local clone of your github repo and  then push that up to your repo.
On the local clone, checkout the `main` branch. Use `git remote add upstream <networkx repo url>`. Then `git fetch upstream main` and finally git merge the upstream main branch to your main branch.  Many files will have a conflict and you'll have to go through one by one to fix the conflicting parts of the code.  Once the git merge is completed you push to your repository.

Another approach would be to `git reset` your main branch back far enough to remove the commits you have made to your main branch. For example, if they are the most recent 15 commits, then 
```
git checkout -b save_15_changes
git checkout main
git reset HEAD~15 --hard
git pull upstream main
# make sure all is OK
git push origin main
```

**Reply by Lukong123** [reply]
Thanks a lot sir, using git reset fixed the problem.

**Comment by sofia-b-chagas** [comment]
Hi guys! This project became one of which I'm most interested to contribute in this year's Outreachy season

Since contribution period already begun, I was wondering if this discussion page will still be used for Outreachy applicants and if yes, if I can start to adress some of the issues listed and maybe contact a mentor

Thank you!

**Reply by MridulS** [reply]
Hey @sofia-b-chagas! The contribution period hasn't started just yet (the date was moved to 8th October)! We appreciate the enthusiasm :) We will start a new discussion for round 2 this year (this discussion thread will be archived).
